app/src/components/ConversationHistory.tsx

```diff
@@ -12,50 +14,57 @@ export interface Conversation {
 export default function ConversationHistory({
     supabaseClient,
     handleClose,
-    fetchLastConversation,
+    setConversationId,
 }: {
     supabaseClient: SupabaseClient;
     handleClose: () => void;
-    fetchLastConversation: (conversationId: number) => void;
+    setConversationId: (id: number) => void;
 }) {
-    const [conversations, setConversations] = useState<Conversation[]>([]);
+    const queryClient = useQueryClient();
+    
+    const deleteConversation = useMutation({
+        mutationFn: async (conversationId: number) => {
+            const allConversations = queryClient.getQueryData<Conversation[]>(['get-all-conversations']);
```

Comment:
nitpick: The convention is to have `conversations` as key. This will not overlap where you fetch by id, because id is part of the key.

Reply:
Oh that makes sense, should I just get rid of the 'getAll' or the 'get' from the variable name too you think?

Reply:
I'd just do `conversations`. Look here https://tanstack.com/query/v4/docs/framework/react/quick-start